src/use-swr.ts

```diff
@@ -551,29 +553,31 @@ export function useSWRHandler<Data = any, Error = any>(
   useIsomorphicLayoutEffect(() => {
     let timer: any = 0
 
-    function nextTick() {
-      if (refreshInterval) {
-        timer = setTimeout(tick, refreshInterval)
+    function next(force?: boolean) {
+      // We only start next interval if `refreshInterval` is not 0, and:
+      // - `force` is true, which is the start of polling
+      // - or `timer` is not 0, which means the effect wasn't canceled
+      if (refreshInterval && (timer || force)) {
```

Comment:
How about using `timer = -1` as initial value, then check `timer && timer !== -1` here, then we don't need `force` arg. Anyway it's not very blocking one

Reply:
Good idea! I'll use `-1` to mark the polling as stopped, thanks!